python/mxnet/gluon/block.py

```diff
@@ -1026,6 +1030,53 @@ def _call_cached_op(self, *args):
             out = [out]
         return _regroup(out, self._out_format)
 
+    def optimize_for(self, x, *args, backend=None, backend_opts=None, **kwargs):
+        """Activates or deactivates :py:class:`HybridBlock` s recursively. Has no effect on
+        non-hybrid children.
+
+        Parameters
+        ----------
+        x : NDArray
+            first input to model
+        *args : NDArray
+            other inputs to model
+        backend : str
+            The name of backend, as registered in `SubgraphBackendRegistry`, default None
+        backend_opts : dict of user-specified options to pass to the backend for partitioning, optional
+            Passed on to `PrePartition` and `PostPartition` functions of `SubgraphProperty`
+        static_alloc : bool, default False
+            Statically allocate memory to improve speed. Memory usage may increase.
+        static_shape : bool, default False
+            Optimize for invariant input shapes between iterations. Must also
+            set static_alloc to True. Change of input shapes is still allowed
+            but slower.
+        """
+
+        # do hybrize API call
+        self.hybridize(True, backend, backend_opts, **kwargs)
```

Comment:
Could you elaborate on your choice of `optimize_for` calling `hybridize`, instead of requiring users to call `hybridize` and then `optimize_for`?

Reply:
The meaning of `optimize_for` API in Symbol class is to partition the model. In Gluon to partition we have to hybridize to build the symbol, and then partition it. So it made sense to integrate that inside as a single step for the user. 

GluonCV does a similar thing with its `export_block` API where they call hybridize under the hood instead to provide a better user experience with a single API call. 
https://github.com/dmlc/gluon-cv/blob/9921323263d3698ce3e03a9eb4ada208b7b8bce3/gluoncv/utils/export_helper.py#L129-L133

Reply:
what if user calls hybridize and then call optimize_for? nothing will happen or it throws on unrelated error message?

Reply:
calling `hybridize` and then `optimize_for` is fine. Since `optimize_for` will just call `hybridize` again internally anyway.

Calling `optimize_for` and then `hybridize` though will reset the cachedOp since `hybridize` calls `_clear_cached_op`. So not a problem per-se. User just needs to realize what they are doing.